ui/src/variables/components/VariableImportOverlay.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {PureComponent} from 'react'
+import {withRouter, WithRouterProps} from 'react-router'
+import {connect} from 'react-redux'
+
+// Components
+import ImportOverlay from 'src/shared/components/ImportOverlay'
+
+// Actions
+import {
+  createVariableFromTemplate as createVariableFromTemplateAction,
+  getVariables as getVariablesAction,
+} from 'src/variables/actions'
+import {notify as notifyAction} from 'src/shared/actions/notifications'
+
+// Types
+import {AppState, Organization} from 'src/types'
+
+interface DispatchProps {
+  getVariables: typeof getVariablesAction
+  createVariableFromTemplate: typeof createVariableFromTemplateAction
+  notify: typeof notifyAction
+}
+
+interface StateProps {
+  org: Organization
+}
+
+interface OwnProps extends WithRouterProps {
+  params: {orgID: string}
+}
+
+type Props = DispatchProps & OwnProps & StateProps
+
+class VariableImportOverlay extends PureComponent<Props> {
+  public render() {
+    return (
+      <ImportOverlay
+        onDismissOverlay={this.onDismiss}
+        resourceName="Variable"
+        onSubmit={this.handleImportVariable}
+      />
+    )
+  }
+
+  private onDismiss = () => {
+    const {router} = this.props
+
+    router.goBack()
+  }
+
+  private handleImportVariable = async (
+    uploadContent: string,
+    orgID: string
+  ): Promise<void> => {
+    const {createVariableFromTemplate, getVariables} = this.props
+
+    const template = JSON.parse(uploadContent)
+    await createVariableFromTemplate(template, orgID)
+
+    await getVariables()
```

Comment:
I think it would be better to dispatch the `SetVariable` action from `createVariableFromTemplate` thunk, rather than having an api call to get all variables. What do you think?

Reply:
the problem with this is that other variables can be included in this template. If i have a variable that uses another variable in my query, then when i import this variable, both should be created. Therefore just setting one variable is not enough. but if we just fetch these variables again, we will get all variables created.

Reply:
Okay, makes sense! Thanks :)